planning/behavior_path_planner/autoware_behavior_path_lane_change_module/include/autoware/behavior_path_lane_change_module/utils/calculation.hpp

```diff
@@ -40,6 +40,23 @@ double calc_ego_dist_to_terminal_end(const CommonDataPtr & common_data_ptr);
 double calc_dist_from_pose_to_terminal_end(
   const CommonDataPtr & common_data_ptr, const lanelet::ConstLanelets & lanes,
   const Pose & src_pose);
+
+/**
+ * @brief Calculates the distance to last fit width position along the lane
+ *
+ * This function computes the distance from the current ego position to the last
+ * position along the lane where the ego foot prints stays within the lane
+ * boundaries.
+ *
+ * @param lanelets current ego lanelets
+ * @param src_pose source pose to calculate distance from
+ * @param common_param common parameters used in behavior path planner.
+ * @param margin additional margin for checking lane width
+ * @return distance to last fit width position along the lane
+ */
+double calc_dist_to_last_fit_width(
+  const lanelet::ConstLanelets lanelets, const Pose & src_pose,
+  const BehaviorPathPlannerParameters & common_param, const double margin = 0.1);
```

Comment:
I would suggest using `bpp_param` instead to avoid confusion with `common_param_ptr`  
```suggestion
  const lanelet::ConstLanelets lanelets, const Pose & src_pose,
  const BehaviorPathPlannerParameters & bpp_param, const double margin = 0.1);
```